sample-package-core/src/test/java/org/nuxeo/sample/core/TestCustomBulkAction.java

```diff
@@ -0,0 +1,56 @@
+package org.nuxeo.sample.core;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.nuxeo.ecm.core.bulk.message.BulkStatus.State.COMPLETED;
+import static org.nuxeo.sample.core.CustomBulkAction.ACTION_NAME;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.CoreBulkFeature;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import javax.inject.Inject;
+
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, CoreBulkFeature.class })
+@Deploy({ "org.nuxeo.sample.core.sample-package-core" })
+public class TestCustomBulkAction {
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Inject
+    protected BulkService service;
+
+    @Inject
+    protected CoreSession session;
+
+    @Test
+    public void testAction() {
+
+        // Change NXQL to retrieve the expected test documents
+        String nxql = "SELECT * FROM Document WHERE ecm:isTrashed = 0";
+
+        BulkCommand command = new BulkCommand.Builder(ACTION_NAME, nxql,
+        session.getPrincipal().getName()).repository(session.getRepositoryName()).build();
+        String commandId = service.submit(command);
+
+        txFeature.nextTransaction();
```

Comment:
it will work because the test feature is waiting for async processing but this is error-prone you could think that commands are part of the transaction. it is better and faster to wait explicitly for the submitted command:
```   assertTrue(bulkService.await(commandId, Duration.ofSeconds(20)));```

Reply:
Thx, good catch. Code changed accordingly